app/screens/main/Main.js

```diff
@@ -99,15 +109,46 @@ class Main extends Component<Props, State> {
     );
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    const { accounts } = this.props;
+    if (prevProps.accounts && !accounts) {
+      this.navToAuthAndLogout();
+    }
+  }
+
   handleSideMenuPress = ({ index }: { index: number }) => {
-    const { history } = this.props;
-    const newPath = sideMenuItems[index].path;
+    const { history, accounts } = this.props;
+    const newPath: ?string = sideMenuItems[index].path;
+    const isLocationLocalNode = newPath && newPath.includes('/local-node');
+    if (!(accounts || isLocationLocalNode) || newPath === '/') {
```

Comment:
(!accounts && !isLocationLocalNode) || newPath === '/'  is much more readable

Reply:
OK.